extensions/wikia/PotentialMemberPageExperiments/scripts/entry-point-experiment.js

```diff
@@ -31,16 +33,42 @@ require([
 	}
 
 	function addEntryPoint(resources) {
-		loader.processStyle(resources.styles);
+		var group = abTest.getGroup(experimentName);
 
-		var templateData = {
-			bannerType: 'block-top'
-		};
+		loader.processStyle(resources.styles);
 
-		$banner = $(mustache.render(resources.mustache[0], templateData));
-		$banner.insertAfter($('.header-container'))
+		$banner = $(mustache.render(template, { bannerType: group.toLowerCase() }))
 			.on('click', '.pmp-entry-point-button', onEntryPointClick)
 			.on('click', '.pmp-entry-point-close', close);
+
+		switch (group) {
+			case 'TOP': addEntryPointTop(); break;
+			case 'IN_ARTICLE': addEntryPointInArticle(); break;
```

Comment:
These should be formatted over multiple lines.

Also, I think it would be cleaner to use a more standard strategy pattern here and set the strategy in the init method that will be then used in a standard way throughout, but since it's an experiment, that's just a nice to have.